taichi/program/program.cpp

```diff
@@ -310,11 +313,11 @@ void Program::check_runtime_error() {
   auto tlctx = llvm_context_host.get();
   auto runtime_jit_module = tlctx->runtime_jit_module;
   runtime_jit_module->call<void *>("runtime_retrieve_error_code", llvm_runtime);
-  auto error_code = runtime_jit_module->fetch_result<int64>();
+  auto error_code = fetch_result<int64>(taichi_result_buffer_error_id);
   if (error_code) {
     runtime_jit_module->call<void *>("runtime_retrieve_error_message",
                                      llvm_runtime);
-    auto error_message = runtime_jit_module->fetch_result<char *>();
+    auto error_message = fetch_result<char *>(taichi_result_buffer_error_id);
     if (error_code == 1) {
       TI_ERROR("Assertion failure: {}", error_message);
     } else {
```

Comment:
So, it's actually:
```
union {
  int64 error_code;
  char *error_message;
};
```
?
They why not just `if (error_message != nullptr)`?

Reply:
`runtime_retrieve_error_message` should set the pointer. It's not a `union`.